### PR TITLE
Type errors - part 1

### LINKED ISSUE #285: Sentry errors - TypeError
**Additional Description**
In the sentry io we see:
https://sentry.io/organizations/bcc-hj/issues/?project=5831770

**Acceptance Criteria**
1. Remove that error

### PR DESCRIPTION
**Pull request by JakubC-projects**
closes #285 

it's supposed to fix those:
https://sentry.io/organizations/bcc-hj/issues/2483484218/?project=5831770&query=is%3Aunresolved
https://sentry.io/organizations/bcc-hj/issues/2483368059/?project=5831770&query=is%3Aunresolved

### CODE REVIEW

File: client/src/components/input-textarea-field.vue [hunk 1]
```diff
@@ -21,23 +21,28 @@ export default defineComponent({
...
     },
     methods: {
         clickEnter(){
             if(this.enableNewLine)
                 this.model += '\n'
+        },
+        adjustHeight(){
+            setTimeout(() => {
+                const element = this.$refs.textArea as HTMLFormElement
+                if(!element)
```
Comment (JakubC-projects) [inline client/src/components/input-textarea-field.vue:41]: This solves one of the errors

File: client/src/views/ThankYou.vue [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <div class="max-w-5xl mx-auto" >
+    <div v-if="!noParams" class="max-w-5xl mx-auto">
```
Comment (JakubC-projects) [inline client/src/views/ThankYou.vue:2]: this solves the other error

File: client/src/router/routes.ts [hunk 3]
```diff
@@ -4,54 +4,54 @@ export const routes: Array<RouteRecordRaw> = [
     {
         path: '/',
         name: 'Home',
-        component: async() => require('../views/Home.vue'),
+        component: () => import('../views/Home.vue' as string),
```
Comment (JakubC-projects) [inline client/src/router/routes.ts:7]: This probably should be in another PR, but the way it works it is allows for lazy loading of the views, so those components would get loaded only when actually necessary, thus allowing the homepage to load faster

File: client/src/components/poll-results.vue [hunk 4]
```diff
@@ -44,7 +44,7 @@ export default defineComponent({
         }
     },
     async created(){
-        this.generateSortedOptions();
+        this.generateSortedOptions()
```
Comment (JakubC-projects) [inline client/src/components/poll-results.vue:47]: This file is just some linting and small refactor

File: client/src/router/routes.ts (hunk 3 above)
Comment (JakubC-projects) [reply client/src/router/routes.ts:7]: The 'as string' is because my editor was underlining it without that. The code works without it too.

Comment (StevenMalaihollo) [reply client/src/router/routes.ts:7]: How is this different from using the require function? I thought this already does what you described here.

Do you have some documentation to back this?

Comment (JakubC-projects) [reply client/src/router/routes.ts:7]: https://router.vuejs.org/guide/advanced/lazy-loading.html#grouping-components-in-the-same-chunk

Comment (JakubC-projects) [reply client/src/router/routes.ts:7]: The require statement loads the code synchronously and import function actually returns a promise.

File: client/src/components/poll-results.vue [hunk 5]
```diff
@@ -120,30 +120,28 @@ export default defineComponent({
...
         },
         changeBars(data: PollResult){
             if(data.pollId !== this.poll._key)
                 return
-            for(const ans in data.answerCount){
-                if(data.answerCount.hasOwnProperty(ans)){
-                    this.sortedOptions[ans].count = data.answerCount[ans];
-                }
-            }
+            Object.keys(this.sortedOptions).forEach((ans: string) =>{
```
Comment (StevenMalaihollo) [inline client/src/components/poll-results.vue:142]: I think the old version is more reliable. What if the poll-result change event comes in before the sortedOptions are generated

**Review by StevenMalaihollo — APPROVED** [review]
Pr looks good, just the one changeBars function change that I don't really understand
